t3/main.py

```diff
@@ -651,27 +651,26 @@ def determine_species_to_calculate(self) -> bool:
         """
         species_keys = list()
 
-        rmg_species, rmg_reactions = self.load_species_and_reactions_from_chemkin_file()
-        self.logger.info(f'This RMG model has {len(rmg_species)} species and {len(rmg_reactions)} reactions in its core.')
+        self.rmg_species, self.rmg_reactions = self.load_species_and_reactions_from_chemkin_file()
+        self.logger.info(f'This RMG model has {len(self.rmg_species)} species and {len(self.rmg_reactions)} reactions in its core.')
 
         if self.t3['options']['all_core_species']:
-            for species in rmg_species:
+            for species in self.rmg_species:
                 if self.species_requires_refinement(species=species):
                     species_keys.append(self.add_species(species=species, reasons=['All core species']))
         else:
             # SA observables
             for input_species in self.rmg['species']:
                 if input_species['observable'] or input_species['SA_observable']:
                     species_keys.append(self.add_species(
-                        species=get_species_by_label(input_species['label'], rmg_species ),
+                        species=get_species_by_label(input_species['label'], self.rmg_species),
```

Comment:
Maybe make `get_species_by_label` a method, so it has access to `self.rmg_species`?